### DISCUSSION TITLE
[QUESTION] Smooth animation?

### DISCUSSION THREAD

**Discussion by ziriax** [mono/SkiaSharp#1611] [Q&A]
When I was making PRs, I noticed that the animation loop code in the demo gallery was not waiting for a "vertical blank interrupt", e.g. an event occurring at the refresh rate of the screen, aka vsync

For example, in WPF one can use the `CompositionTarget.Rendering` [event for this](https://docs.microsoft.com/en-us/dotnet/api/system.windows.media.compositiontarget.rendering?view=netcore-3.1)

I don't think SkiaSharp offers a platform independent interface for this?

**Comment by mattleibow** [comment]
Not exactly. This is a platform feature, so on iOS you use the `CADisplayLink` and on Android you would use the `Choreographer`

The Forms views should be using this for the GL views. But, since the other views do not have a "render loop" operation, the implementation actually needs to be implemented via the framework. Right now an option is the device timers. But this can probably be improved.

Did you maybe mean something else?

**Reply by ziriax** [reply]
Thanks for the reply.  

As an example, on web platforms (javascript), one uses [`requestAnimationFrame`](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) for this.

A platform feature it is then, no abstraction around it yet.
